torchgen/api/autograd.py

```diff
@@ -316,53 +316,53 @@ def is_reference_for_foreach(
     f: NativeFunction,
     function_schema: FunctionSchema,
 ) -> bool:
-    return (
-        f.func.name.name.base.split("_foreach_")[-1] == function_schema.name.name.base
-        and not function_schema.name.name.inplace
-        and all(
-            ref_arg.type in (arg.type, getattr(arg.type, "elem", None))
-            for arg, ref_arg in zip(
-                f.func.arguments.flat_non_out,
-                function_schema.arguments.flat_non_out,
-            )
+    return f.func.name.name.base.split("_foreach_")[
+        -1
+    ] == function_schema.name.name.base and all(
```

Comment:
haha this is an interesting line break, took me a second to parse this
any way we could keep it on the same line as before?

Reply:
Also we should probably ignore the `and not function_schema.name.name.inplace` condition only  in the case of zero_?

We could have a single set that is used both here and below in `gen_foreach_derivativeinfo`.